From bde677baf2d1ebc3df47061105d3c982ae0cf1bd Mon Sep 17 00:00:00 2001
From: Ben Shelton <benjamin.h.shelton@intel.com>
Date: Fri, 11 Mar 2016 09:36:42 -0600
Subject: [PATCH 4/6] pci-driver: HACK: don't allocate additional bridge window
 space for hotplug

On the frio boards, when the PCI link is set to 4x2 mode, the kernel allocates
hotplug space behind all of the PCI bridge windows, which uses up enough of the
available 32-bit PCI window address space that the X9/XLF device is not able to
enumerate all its BARs.

Since PCI hotplug is not expected on this system, don't allocate hotplug space
for any bridges.  This enables all the BARs to fit.

This commit should be reverted as soon as a better solution is available.

Signed-off-by: Ben Shelton <benjamin.h.shelton@intel.com>
---
 drivers/pci/setup-bus.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/pci/setup-bus.c b/drivers/pci/setup-bus.c
index 4225699..e8ab8c4 100644
--- a/drivers/pci/setup-bus.c
+++ b/drivers/pci/setup-bus.c
@@ -1264,10 +1264,12 @@ void __pci_bus_size_bridges(struct pci_bus *bus, struct list_head *realloc_head)
 
 	case PCI_CLASS_BRIDGE_PCI:
 		pci_bridge_check_ranges(bus);
+#if 0
 		if (bus->self->is_hotplug_bridge) {
 			additional_io_size  = pci_hotplug_io_size;
 			additional_mem_size = pci_hotplug_mem_size;
 		}
+#endif
 		/* Fall through */
 	default:
 		pbus_size_io(bus, realloc_head ? 0 : additional_io_size,
-- 
2.7.4

